src/SAML2/SubjectQuery.php

```diff
@@ -77,21 +78,23 @@ private function parseSubject(\DOMElement $xml): void
     /**
      * Retrieve the NameId of the subject in the query.
      *
-     * @return \SAML2\XML\saml\NameID|null The name identifier of the assertion.
+     * @return \SAML2\XML\saml\NameID The name identifier of the assertion.
      */
-    public function getNameId(): ?NameID
+    public function getNameId(): NameID
     {
+        Assert::notEmpty($this->nameId);
```

Comment:
I was playing a bit before with some examples, and PHP definitely throws an exception when you try to return a type that doesn't match the signature of the function. However, I don't know how psalm would deal with that case. It's probably a good idea to be explicit here anyway.